src/components/selection_input/selection_input.ts

```diff
@@ -62,6 +78,7 @@ interface Props {
   class?: string;
   onSelectionChanged?: (ranges: string[]) => void;
   onSelectionConfirmed?: () => void;
+  postProcessSelection?: (ranges: (RangeInputValue & { isFocused: boolean })[], id: string) => void;
```

Comment:
the SelectionInput API is already SOTG. This doesn't make it easier.
IMO we should check how `onSelectionChanged` and `onSelectionConfirmed` are used and clarify the situation.
Currently, the state is duplicated inside the plugin and in the parents (check `onSelectionChanged`)

I think we can explore this behavior:
1. when not focused: just display ranges from it's props (no internal state)
2. when focused: display it's internal state
3. when confirmed: send the changes to its parent (with onSelectionConfirmed), clean it's internal state and get back to 1.

That way, I think `spreadRange` would not be needed at all. We would just rely on the existing business logic of charts/CF which, I assume, already do the very same job of splitting ranges

Reply:
Ok, sorry, it can't work like what I proposed. From what I see now, the "Save" button in the CF side panel takes over the "Confirm" button, so the state must be held in the parent component.
This API requires a bit more thoughts then that